README.md
# Quick Search Plus


---

## 介绍
这是一个快速搜索插件
+ 快速搜索必应
+ 快速查看热搜
+ 快速搜索ChatGpt3.5/4.0


---

## 普通用法

按下f1，输入quick Search然后回车！ 

---

## 进阶用法

### Chatgpt3.5
ctrl + alt + G快捷键使用 
### Chatgpt4.0
内测中...

---

## 致谢
+ 感谢一之哥哥=>API来源[Go 题库](https://q.icodef.com/)，大家多多支持


---

## 关于

Author：[bigonion]()
e-Mail：bigonion@bigonion.cn  
NameSpace: [大聪花家](https://bigonion.cn)   
Origin: [大聪花的博客](https://bigonion.cn/blog)    
Powered by [markdown在线编辑](https://md.bigonion.cn/)   

声明：未经本人同意，禁止转载、搬运、抄袭！ 

CHANGELOG.md
# Change Log

All notable changes to the "onion-vscode" extension will be documented in this file.

Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how to structure this file.

## [Unreleased]

- Initial release

test.js
const http = require('http');  // 引入 HTTP 模块

const data = 'id=1&name=Louis';  // 要发送的内容
// 使用 request 方法发送 POST 请求
const req = http.request({
  hostname: 'localhost',
  path: '/php/test/api.php',
  port: 80,
  method: 'POST',
}, res => {
  // 如果状态码不是 200 就输出状态码
  if (res.statusCode !== 200) console.log(res.statusCode);

  res.on('data', d => {
    // 把接收到的内容转为字符串在控制台输出
    console.log(d.toString())
  });
});

// 设置要发送的内容
req.write(data);
// 如果出错就在控制台输出错误信息
req.on('error', err => {
  console.log(err);
});

req.end();  // 结束

out/test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const axios_1 = require("axios");
async function name(params) {
    const res = await axios_1.default.get("https://bigonion.cn");
    console.log(res);
}
name(1);
//# sourceMappingURL=test.js.map

src/extension.ts
import * as vscode from "vscode";
import axios, { AxiosHeaders } from "axios";
import * as https from "https";
import * as http from "http";
import "./extension";
import * as jwt from "jsonwebtoken";
// 解决https证书报错
// axios在vscode里不行
const httpsAgent = new https.Agent({ rejectUnauthorized: false });
const httpAgent = new http.Agent({});
// let ques: string = "hello";

export async function activate(context: vscode.ExtensionContext) {
  try {
    // vscode.window.showInformationMessage(a);
  } catch (e) {
    console.log(e);
  }

  let quickSearchCommon = vscode.commands.registerCommand(
    "quick-search-plus.qs",
    async () => {
      // choices
      const list: string[] = ["快速搜索", "热搜", "About Me"];
      const inputs: any = await vscode.window.showQuickPick(list, {
        matchOnDetail: true,
      });
      if (inputs === "About Me") {
        const myWeb: any = "https://bigonion.cn";
        vscode.env.openExternal(myWeb);
      }
      if (inputs === "快速搜索") {
        vscode.window
          .showInputBox({
            // 这个对象中所有参数都是可选参数
            password: false, // 输入内容是否是密码
            ignoreFocusOut: true, // 默认false，设置为true时鼠标点击别的地方输入框不会消失
            placeHolder: "你到底想输入什么？", // 在输入框内的提示信息
            prompt: "赶紧输入，不输入就关掉！", // 在输入框下方的提示信息
            // validateInput:function(text){return text;} // 对输入内容进行验证并返回
          })
          .then(function (msg: any) {
            console.log("用户输入：" + msg);
            if (msg) {
              msg = "https://cn.bing.com/search?q=" + msg;
              vscode.env.openExternal(msg);
            }
          });
      }
      if (inputs === "热搜") {
        axios({
          url: "https://tenapi.cn/resou/",
          method: "POST",
        })
          .then((e: any) => {
            // console.log(e.data.list);
            let resouList = [];
            for (let i = 0; i <= 20; i++) {
              resouList.push(e.data.list[i].name);
            }
            return vscode.window.showQuickPick(resouList);
          })
          .then((e: any) => {
            if (e) {
              e = "https://cn.bing.com/search?q=" + e;
              vscode.env.openExternal(e);
            }
          });
      }
    }
  );
  let quickGPT = vscode.commands.registerCommand(
    "quick-search-plus.qGPT",
    async () => {
      vscode.window
        .showInputBox({
          password: false, // 输入内容是否是密码
          ignoreFocusOut: false, // 默认false，设置为true时鼠标点击别的地方输入框不会消失
          placeHolder: "OPENAI READY", // 在输入框内的提示信息
          prompt: "请输入你的问题", // 在输入框下方的提示信息
          // validateInput:function(text){return text;} // 对输入内容进行验证并返回
        })
        .then((e) => {
          try {
            // e 是问题
            if (e) {
              thinkEvent(true)//思考
              askGPT(e, "gpt-3.5-turbo-0301").then((e: any) => {
                // console.log(e.data.choices[0]);
                vscode.window.showInformationMessage(
                  e.data.choices[0].message.content
                );
                thinkEvent(false)//结束思考
              });
            }
          } catch (error: any) {
            console.log(error);
          }
        });
    }
  );
  let quickGPT4 = vscode.commands.registerCommand(
    "quick-search-plus.qGPT4.0",
    async () => {
      let _token_: any = getConfigValue("GPT4.0token");
      try {
        if (!_token_ || !normalDecode(_token_, "iloveJavascript&")) {
          vsAlert("请填入正确的token");
          return;
        }
      } catch (error) {
        vsAlert("请填入正确的token");
        return;
      }

      vscode.window
        .showInputBox({
          password: false, // 输入内容是否是密码
          ignoreFocusOut: false, // 默认false，设置为true时鼠标点击别的地方输入框不会消失
          placeHolder: "OPENAI READY", // 在输入框内的提示信息
          prompt: "请输入你的问题", // 在输入框下方的提示信息
          // validateInput:function(text){return text;} // 对输入内容进行验证并返回
        })
        .then((e) => {
          try {
            if (e) {
              thinkEvent(true)//思考
              askGPT(e, normalDecode(_token_, "iloveJavascript&").type).then(
                (e: any) => {
                  // console.log(e.data.choices[0]);
                  vscode.window.showInformationMessage(
                    e.data.choices[0].message.content
                  );
                  thinkEvent(false)//结束思考
                }
              );
            }
          } catch (error: any) {
            console.log(error);
          }
        });
    }
  );

  // 注册命令区
  context.subscriptions.push(quickSearchCommon);
  context.subscriptions.push(quickGPT);
  context.subscriptions.push(quickGPT4);
}

// This method is called when your extension is deactivated
export function deactivate() {}

function askGPT(ques: string, token?: string): Promise<any> {
  // token=token?"gpt-3.5-turbo-0301":""
  return axios({
    headers: {
      // eslint-disable-next-line @typescript-eslint/naming-convention
      "Content-Type": "application/json",
    },
    httpsAgent: httpsAgent,
    method: "POST",
    url: "https://q.icodef.com/v1/chat/completions",
    data: `{
  "token": "6spM5IRmrXskQHMA",
  "model":"${token}",
  "stream":false,
  "ret_usage":false,
  "messages": [
    {
      "role": "user",
      "content": "${ques}",
      "createtime": ${Date.now()},
      "name": "Onion"
    }
  ],
  "use_context": true
}
`,
  });
}

async function axiosGPT(ques: string) {
  const a = await axios({
    url: "http://api.bigonion.cn",
    method: "POST",
    data: { prompt: `${ques}` },
    httpsAgent: httpsAgent,
    httpAgent: httpAgent,
  });
  return a;
}

/**
 * @description getVscode Config
 * @param name
 * @returns config value
 */
function getConfigValue(name: string): any {
  return vscode.workspace.getConfiguration("bigonion").get(name);
}

/**
 * @description JWT
 */
function normalDecode(token: string, secret: string): any {
  let decoded = jwt.verify(token, secret);
  // console.log(decoded)
  return decoded;
}

function vsAlert(e: any) {
  vscode.window.showInformationMessage(e);
}

function thinkEvent(startOrEnd:boolean){
  let dots = "...";
  // start 1 end 0
  if(startOrEnd){
    vscode.window.setStatusBarMessage("chatGPT is thinking" + dots);
  }else{
    vscode.window.setStatusBarMessage(" ");
  }
}


out/extension.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deactivate = exports.activate = void 0;
const vscode = require("vscode");
const axios_1 = require("axios");
const https = require("https");
const http = require("http");
require("./extension");
const jwt = require("jsonwebtoken");
// 解决https证书报错
// axios在vscode里不行
const httpsAgent = new https.Agent({ rejectUnauthorized: false });
const httpAgent = new http.Agent({});
// let ques: string = "hello";
async function activate(context) {
    try {
        // vscode.window.showInformationMessage(a);
    }
    catch (e) {
        console.log(e);
    }
    let quickSearchCommon = vscode.commands.registerCommand("quick-search-plus.qs", async () => {
        // choices
        const list = ["快速搜索", "热搜", "About Me"];
        const inputs = await vscode.window.showQuickPick(list, {
            matchOnDetail: true,
        });
        if (inputs === "About Me") {
            const myWeb = "https://bigonion.cn";
            vscode.env.openExternal(myWeb);
        }
        if (inputs === "快速搜索") {
            vscode.window
                .showInputBox({
                // 这个对象中所有参数都是可选参数
                password: false,
                ignoreFocusOut: true,
                placeHolder: "你到底想输入什么？",
                prompt: "赶紧输入，不输入就关掉！", // 在输入框下方的提示信息
                // validateInput:function(text){return text;} // 对输入内容进行验证并返回
            })
                .then(function (msg) {
                console.log("用户输入：" + msg);
                if (msg) {
                    msg = "https://cn.bing.com/search?q=" + msg;
                    vscode.env.openExternal(msg);
                }
            });
        }
        if (inputs === "热搜") {
            (0, axios_1.default)({
                url: "https://tenapi.cn/resou/",
                method: "POST",
            })
                .then((e) => {
                // console.log(e.data.list);
                let resouList = [];
                for (let i = 0; i <= 20; i++) {
                    resouList.push(e.data.list[i].name);
                }
                return vscode.window.showQuickPick(resouList);
            })
                .then((e) => {
                if (e) {
                    e = "https://cn.bing.com/search?q=" + e;
                    vscode.env.openExternal(e);
                }
            });
        }
    });
    let quickGPT = vscode.commands.registerCommand("quick-search-plus.qGPT", async () => {
        vscode.window
            .showInputBox({
            password: false,
            ignoreFocusOut: false,
            placeHolder: "OPENAI READY",
            prompt: "请输入你的问题", // 在输入框下方的提示信息
            // validateInput:function(text){return text;} // 对输入内容进行验证并返回
        })
            .then((e) => {
            try {
                // e 是问题
                if (e) {
                    thinkEvent(true); //思考
                    askGPT(e, "gpt-3.5-turbo-0301").then((e) => {
                        // console.log(e.data.choices[0]);
                        vscode.window.showInformationMessage(e.data.choices[0].message.content);
                        thinkEvent(false); //结束思考
                    });
                }
            }
            catch (error) {
                console.log(error);
            }
        });
    });
    let quickGPT4 = vscode.commands.registerCommand("quick-search-plus.qGPT4.0", async () => {
        let _token_ = getConfigValue("GPT4.0token");
        try {
            if (!_token_ || !normalDecode(_token_, "iloveJavascript&")) {
                vsAlert("请填入正确的token");
                return;
            }
        }
        catch (error) {
            vsAlert("请填入正确的token");
            return;
        }
        vscode.window
            .showInputBox({
            password: false,
            ignoreFocusOut: false,
            placeHolder: "OPENAI READY",
            prompt: "请输入你的问题", // 在输入框下方的提示信息
            // validateInput:function(text){return text;} // 对输入内容进行验证并返回
        })
            .then((e) => {
            try {
                if (e) {
                    thinkEvent(true); //思考
                    askGPT(e, normalDecode(_token_, "iloveJavascript&").type).then((e) => {
                        // console.log(e.data.choices[0]);
                        vscode.window.showInformationMessage(e.data.choices[0].message.content);
                        thinkEvent(false); //结束思考
                    });
                }
            }
            catch (error) {
                console.log(error);
            }
        });
    });
    // 注册命令区
    context.subscriptions.push(quickSearchCommon);
    context.subscriptions.push(quickGPT);
    context.subscriptions.push(quickGPT4);
}
exports.activate = activate;
// This method is called when your extension is deactivated
function deactivate() { }
exports.deactivate = deactivate;
function askGPT(ques, token) {
    // token=token?"gpt-3.5-turbo-0301":""
    return (0, axios_1.default)({
        headers: {
            // eslint-disable-next-line @typescript-eslint/naming-convention
            "Content-Type": "application/json",
        },
        httpsAgent: httpsAgent,
        method: "POST",
        url: "https://q.icodef.com/v1/chat/completions",
        data: `{
  "token": "6spM5IRmrXskQHMA",
  "model":"${token}",
  "stream":false,
  "ret_usage":false,
  "messages": [
    {
      "role": "user",
      "content": "${ques}",
      "createtime": ${Date.now()},
      "name": "Onion"
    }
  ],
  "use_context": true
}
`,
    });
}
async function axiosGPT(ques) {
    const a = await (0, axios_1.default)({
        url: "http://api.bigonion.cn",
        method: "POST",
        data: { prompt: `${ques}` },
        httpsAgent: httpsAgent,
        httpAgent: httpAgent,
    });
    return a;
}
/**
 * @description getVscode Config
 * @param name
 * @returns config value
 */
function getConfigValue(name) {
    return vscode.workspace.getConfiguration("bigonion").get(name);
}
/**
 * @description JWT
 */
function normalDecode(token, secret) {
    let decoded = jwt.verify(token, secret);
    // console.log(decoded)
    return decoded;
}
function vsAlert(e) {
    vscode.window.showInformationMessage(e);
}
function thinkEvent(startOrEnd) {
    let dots = "...";
    // start 1 end 0
    if (startOrEnd) {
        vscode.window.setStatusBarMessage("chatGPT is thinking" + dots);
    }
    else {
        vscode.window.setStatusBarMessage(" ");
    }
}
//# sourceMappingURL=extension.js.map